tfs/src/main/java/hudson/plugins/tfs/model/AbstractHookEvent.java

```diff
@@ -111,90 +157,10 @@ static JSONObject fromResponseContributors(final List<GitStatus.ResponseContribu
                 if (scmTriggerItem == null) {
                     continue;
                 }
-                for (final SCM scm : scmTriggerItem.getSCMs()) {

```

Comment:
I am not sure about the removal of this code. You mentioned that it was duplicated in runPolling, but I don't see how that affects the loops here?
The first loop makes sure we only looking at Git scm providers
The other loops make sure that the repository matches. 
I don't see this code in runPolling. And even if it is there, you allow runpolling to return false and still get here if there is a commit.
Is there a reason to remove the code other than it being duplicated? Perhaps we should leave it here just in case it serves another purpose.